src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java

```diff
@@ -0,0 +1,390 @@
+        {
+            buildingEntry.setOnGoingDelivery(true);
+            list.add(buildingEntry);
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Check all chests in the worker hut for a required item.
+     * @param is the type of item requested (amount is ignored)
+     * @return true if a stack of that type was found
+     */
+    public boolean isInHut(@Nullable final ItemStack is)
+    {
+        @Nullable final AbstractBuilding building = getBuilding();
+
+        boolean hasItem;
+        if(building != null)
+        {
+            hasItem = isInTileEntity(building.getTileEntity(), is);
+
+            if(hasItem)
+            {
+                return true;
+            }
+
+            for(BlockPos pos : building.getAdditionalCountainers())
+            {
+                TileEntity entity = worldObj.getTileEntity(pos);
+                if(entity instanceof TileEntityChest)
+                {
+                    hasItem = isInTileEntity((TileEntityChest) entity, is);
+
+                    if(hasItem)
+                    {
+                        return true;
+                    }
+                }
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Check for a certain item and return the position of the chest containing it.
+     * @param stack the stack to search for.
+     * @return the position or null.
+     */
+    @Nullable
+    public BlockPos getPositionOfChestWithItemStack(ItemStack stack)
+    {
+        @Nullable final AbstractBuilding building = getBuilding();
+
+        if(building != null)
+        {
+            if(isInTileEntity(building.getTileEntity(), stack))
+            {
+                return building.getLocation();
+            }
+
+            for(BlockPos pos : building.getAdditionalCountainers())
+            {
+                TileEntity entity = worldObj.getTileEntity(pos);
+                if(entity instanceof TileEntityChest && isInTileEntity((TileEntityChest) entity, stack))
+                {
+                    return pos;
+                }
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Check for a certain item and return the position of the chest containing it.
+     * @param tool the tool to search for.
+     * @param minLevel the minLevel of the pickaxe
+     * @return the position or null.
+     */
+    public BlockPos getPositionOfChestWithTool(String tool, int minLevel)
+    {
+        @Nullable final AbstractBuilding building = getBuilding();
+
+        if(building != null)
+        {
+            if((minLevel != -1 && InventoryUtils.isPickaxeInTileEntity(building.getTileEntity(), minLevel)) || InventoryUtils.isToolInTileEntity(building.getTileEntity(), tool))
+            {
+                return building.getLocation();
+            }
+
+            for(BlockPos pos : building.getAdditionalCountainers())
+            {
+                TileEntity entity = worldObj.getTileEntity(pos);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'entity' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)